File: backend/schemas/company.py
### Company schema ###

def company_schema(company) -> dict:
    return {"id": str(company["_id"]), # Pasamos Objeto a String
            "name": company["name"],
            "address": company["address"],
            "city_id": company["city_id"],
            "email": company["email"],
            "phone": company["phone"],
            "webpage": company["webpage"],
            "logo": company["logo"]}


def companies_schema(companies) -> list:
    return [company_schema(company) for company in companies]

File: frontend/src/types/company/company.ts
import { z } from "zod";


export const companySchema = z.object({
    id: z.string(),
    name: z.string(),
    address: z.string(),
    email: z.string().email(),
    phone: z.string(),
    webpage: z.string(),
    logo: z.string(),
    city_id: z.string(),
})


export type Company = z.infer<typeof companySchema>